pkg/ddl/partition.go

```diff
@@ -623,16 +623,83 @@ func buildTablePartitionInfo(ctx sessionctx.Context, s *ast.PartitionOptions, tb
 		}
 	}
 
-	partCols, err := getPartitionColSlices(exprCtx, tbInfo, s)
-	if err != nil {
-		return errors.Trace(err)
+	if len(s.UpdateIndexes) > 0 {
+		updateIndexes := make([]model.UpdateIndexInfo, 0, len(s.UpdateIndexes))
+		dupCheck := make(map[string]struct{})
+		for _, idxUpdate := range s.UpdateIndexes {
+			idxOffset := -1
+			for i := range tbInfo.Indices {
+				if strings.EqualFold(tbInfo.Indices[i].Name.L, idxUpdate.Name) {
+					idxOffset = i
```

Comment:
Should we check is commonHandle here?

Reply:
Not sure, it is checked in the next loop over tbInfo.Indices, and will fail if CommonIndex and global index is set for the same index.